only advance _count if dummy_index is not given

smichr opened the pull request.

cbm755 commented:
I cherry-picked the first two for now.  Thanks for getting involved in this; it seems like a minor thing but details are tricky!

smichr committed:
track whether dummy_index was given

What is a better name than 'given' for the Dummy attribute
indicating whether the dummy_index was explicitly given or
not?

smichr committed:
update sort_key to reflect that there are 3 items

smichr commented:
I ran with your idea of tracking whether the dummy index was given or not.

cbm755 commented on sympy/core/tests/test_symbol.py:
```diff
@@ -30,15 +30,19 @@ def test_Symbol():
     assert Dummy('d', dummy_index=2) == Dummy('d', dummy_index=2)
     assert Dummy('d1', dummy_index=2) != Dummy('d2', dummy_index=2)
+    assert Dummy('d', dummy_index=2, given=1) == Dummy('d', dummy_index=2, given=1)
+    assert Dummy('d', dummy_index=2, given=0) == Dummy('d', dummy_index=2, given=0)
```
hmmm, this example makes me feel this `given` kwarg doesn't add anything...

cbm755 commented:
I don't think `given` should be part of the API for this.  If we want to go this route (not saying I want too), then I think:

````
_given = True if dummy_index is not None else False
````
and for two Dummys to be equal, they must have the same `name`, same `dummy_index` and same `_given`.

With this:
````
>>> d1 = Dummy('d')
>>> s = srepr(d1)
Dummy('d', dummy_index=17)
>>> d2 = Dummy('d', dummy_index=17)
>>> d3 = Dummy('d', dummy_index=17)
>>> d2 == d3   # Good!
True
>>> d1 == d2   # hmmm
False
````
Last result may be unnerving but no worse than we have today.  The rule is simple: any Dummy made in "the usual way" is not equal to any other Dummy.
